src/sentry/web/frontend/accounts.py

```diff
@@ -121,9 +122,18 @@ def recover_confirm(request, user_id, hash):
 
 
 @login_required
+@require_http_methods(["POST"])
 def start_confirm_email(request):
     has_unverified_emails = request.user.has_unverified_emails()
```

Comment:
Should move this check now into the `elif` block you added. Since if you hit the `primary-email` path, this query is unnecessary.

So I'd just do:

```python
if 'primary-email' in request.POST:
    ....
elif request.user.has_verified_emails():
   ....
```